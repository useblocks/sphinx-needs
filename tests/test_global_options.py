from pathlib import Path

import pytest


@pytest.mark.parametrize(
    "test_app",
    [{"buildername": "html", "srcdir": "doc_test/doc_global_options"}],
    indirect=True,
)
def test_doc_global_option(test_app):
    app = test_app
    app.build()
    html = Path(app.outdir, "index.html").read_text()

    assert "global_1" in html
    assert "global_2" in html
    assert "global_3" in html
    assert "global_4" in html
    assert "global_5" in html

    assert "test_global" in html
    assert "1.27" in html
    assert "Test output of dynamic function; need: GLOBAL_ID" in html

    assert "STATUS_IMPL" in html
    assert "STATUS_UNKNOWN" in html
    assert "STATUS_CLOSED" not in html
