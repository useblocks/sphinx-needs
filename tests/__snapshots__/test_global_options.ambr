# serializer version: 1
# name: test_doc_global_option[test_app0]
  dict({
    'layout': dict({
      'default': 'clean_l',
    }),
    'link1': dict({
      'default': list([
        'SPEC_1',
      ]),
    }),
    'link2': dict({
      'default': list([
        'SPEC_1',
      ]),
      'predicates': list([
        tuple(
          'status == "implemented"',
          list([
            'SPEC_2',
            "[[copy('link1')]]",
          ]),
        ),
        tuple(
          'status == "closed"',
          list([
            'SPEC_3',
          ]),
        ),
      ]),
    }),
    'option_1': dict({
      'default': 'test_global',
    }),
    'option_2': dict({
      'default': "[[copy('id')]]",
    }),
    'option_3': dict({
      'predicates': list([
        tuple(
          'status == "implemented"',
          'STATUS_IMPL',
        ),
      ]),
    }),
    'option_4': dict({
      'default': 'STATUS_UNKNOWN',
      'predicates': list([
        tuple(
          'status == "closed"',
          'STATUS_CLOSED',
        ),
      ]),
    }),
    'option_5': dict({
      'default': 'final',
      'predicates': list([
        tuple(
          'status == "implemented"',
          'STATUS_IMPL',
        ),
        tuple(
          'status == "closed"',
          'STATUS_CLOSED',
        ),
      ]),
    }),
    'tags': dict({
      'default': list([
        'd',
      ]),
      'predicates': list([
        tuple(
          'status == "implemented"',
          list([
            'a',
            'b',
          ]),
        ),
        tuple(
          'status == "closed"',
          list([
            'c',
          ]),
        ),
      ]),
    }),
  })
# ---
# name: test_doc_global_option[test_app0].1
  dict({
    'current_version': '',
    'versions': dict({
      '': dict({
        'needs': dict({
          'SPEC_1': dict({
            'docname': 'index',
            'external_css': 'external_link',
            'id': 'SPEC_1',
            'layout': 'clean_l',
            'lineno': 4,
            'link1': list([
              'SPEC_1',
            ]),
            'link1_back': list([
              'SPEC_1',
              'SPEC_2',
              'SPEC_3',
            ]),
            'link2': list([
              'SPEC_2',
              'SPEC_1',
            ]),
            'link2_back': list([
              'SPEC_1',
              'SPEC_3',
            ]),
            'option_1': 'other',
            'option_2': 'SPEC_1',
            'option_3': 'STATUS_IMPL',
            'option_4': 'STATUS_UNKNOWN',
            'option_5': 'STATUS_IMPL',
            'section_name': 'GLOBAL OPTIONS',
            'sections': list([
              'GLOBAL OPTIONS',
            ]),
            'status': 'implemented',
            'tags': list([
              'a',
              'b',
            ]),
            'title': 'Specification 1',
            'type': 'spec',
            'type_name': 'Specification',
          }),
          'SPEC_2': dict({
            'docname': 'index',
            'external_css': 'external_link',
            'id': 'SPEC_2',
            'layout': 'clean_l',
            'lineno': 9,
            'link1': list([
              'SPEC_1',
            ]),
            'link2': list([
              'SPEC_3',
            ]),
            'link2_back': list([
              'SPEC_1',
            ]),
            'option_1': 'test_global',
            'option_2': 'SPEC_2',
            'option_4': 'STATUS_CLOSED',
            'option_5': 'STATUS_CLOSED',
            'section_name': 'GLOBAL OPTIONS',
            'sections': list([
              'GLOBAL OPTIONS',
            ]),
            'status': 'closed',
            'tags': list([
              'c',
            ]),
            'title': 'Specification 2',
            'type': 'spec',
            'type_name': 'Specification',
          }),
          'SPEC_3': dict({
            'docname': 'index',
            'external_css': 'external_link',
            'id': 'SPEC_3',
            'layout': 'clean_l',
            'lineno': 13,
            'link1': list([
              'SPEC_1',
            ]),
            'link2': list([
              'SPEC_1',
            ]),
            'link2_back': list([
              'SPEC_2',
            ]),
            'option_1': 'test_global',
            'option_2': 'SPEC_3',
            'option_4': 'STATUS_UNKNOWN',
            'option_5': 'final',
            'section_name': 'GLOBAL OPTIONS',
            'sections': list([
              'GLOBAL OPTIONS',
            ]),
            'status': 'other',
            'tags': list([
              'd',
            ]),
            'title': 'Specification 3',
            'type': 'spec',
            'type_name': 'Specification',
          }),
        }),
        'needs_amount': 3,
        'needs_defaults_removed': True,
      }),
    }),
  })
# ---
# name: test_doc_global_option_old[test_app0]
  dict({
    'layout': dict({
      'default': 'clean_l',
    }),
    'link1': dict({
      'default': list([
        'SPEC_1',
      ]),
    }),
    'link2': dict({
      'default': list([
        'SPEC_1',
      ]),
      'predicates': list([
        tuple(
          'status == "implemented"',
          list([
            'SPEC_2',
            "[[copy('link1')]]",
          ]),
        ),
        tuple(
          'status == "closed"',
          list([
            'SPEC_3',
          ]),
        ),
      ]),
    }),
    'option_1': dict({
      'default': 'test_global',
    }),
    'option_2': dict({
      'default': "[[copy('id')]]",
    }),
    'option_3': dict({
      'predicates': list([
        tuple(
          'status == "implemented"',
          'STATUS_IMPL',
        ),
      ]),
    }),
    'option_4': dict({
      'default': 'STATUS_UNKNOWN',
      'predicates': list([
        tuple(
          'status == "closed"',
          'STATUS_CLOSED',
        ),
      ]),
    }),
    'option_5': dict({
      'default': 'final',
      'predicates': list([
        tuple(
          'status == "implemented"',
          'STATUS_IMPL',
        ),
        tuple(
          'status == "closed"',
          'STATUS_CLOSED',
        ),
      ]),
    }),
    'tags': dict({
      'default': list([
        'd',
      ]),
      'predicates': list([
        tuple(
          'status == "implemented"',
          list([
            'a',
            'b',
          ]),
        ),
        tuple(
          'status == "closed"',
          list([
            'c',
          ]),
        ),
      ]),
    }),
  })
# ---
# name: test_doc_global_option_old[test_app0].1
  dict({
    'current_version': '',
    'versions': dict({
      '': dict({
        'needs': dict({
          'SPEC_1': dict({
            'docname': 'index',
            'external_css': 'external_link',
            'id': 'SPEC_1',
            'layout': 'clean_l',
            'lineno': 4,
            'link1': list([
              'SPEC_1',
            ]),
            'link1_back': list([
              'SPEC_1',
              'SPEC_2',
              'SPEC_3',
            ]),
            'link2': list([
              'SPEC_2',
              'SPEC_1',
            ]),
            'link2_back': list([
              'SPEC_1',
              'SPEC_3',
            ]),
            'option_1': 'other',
            'option_2': 'SPEC_1',
            'option_3': 'STATUS_IMPL',
            'option_4': 'STATUS_UNKNOWN',
            'option_5': 'STATUS_IMPL',
            'section_name': 'GLOBAL OPTIONS',
            'sections': list([
              'GLOBAL OPTIONS',
            ]),
            'status': 'implemented',
            'tags': list([
              'a',
              'b',
            ]),
            'title': 'Specification 1',
            'type': 'spec',
            'type_name': 'Specification',
          }),
          'SPEC_2': dict({
            'docname': 'index',
            'external_css': 'external_link',
            'id': 'SPEC_2',
            'layout': 'clean_l',
            'lineno': 9,
            'link1': list([
              'SPEC_1',
            ]),
            'link2': list([
              'SPEC_3',
            ]),
            'link2_back': list([
              'SPEC_1',
            ]),
            'option_1': 'test_global',
            'option_2': 'SPEC_2',
            'option_4': 'STATUS_CLOSED',
            'option_5': 'STATUS_CLOSED',
            'section_name': 'GLOBAL OPTIONS',
            'sections': list([
              'GLOBAL OPTIONS',
            ]),
            'status': 'closed',
            'tags': list([
              'c',
            ]),
            'title': 'Specification 2',
            'type': 'spec',
            'type_name': 'Specification',
          }),
          'SPEC_3': dict({
            'docname': 'index',
            'external_css': 'external_link',
            'id': 'SPEC_3',
            'layout': 'clean_l',
            'lineno': 13,
            'link1': list([
              'SPEC_1',
            ]),
            'link2': list([
              'SPEC_1',
            ]),
            'link2_back': list([
              'SPEC_2',
            ]),
            'option_1': 'test_global',
            'option_2': 'SPEC_3',
            'option_4': 'STATUS_UNKNOWN',
            'option_5': 'final',
            'section_name': 'GLOBAL OPTIONS',
            'sections': list([
              'GLOBAL OPTIONS',
            ]),
            'status': 'other',
            'tags': list([
              'd',
            ]),
            'title': 'Specification 3',
            'type': 'spec',
            'type_name': 'Specification',
          }),
        }),
        'needs_amount': 3,
        'needs_defaults_removed': True,
      }),
    }),
  })
# ---
